From f306160ca688acdebf0c460955f71a2a646db250 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 16 Apr 2021 11:04:30 +0200
Subject: [PATCH] Revert "build: Bail with error if non-existant build-type
 specified (#2965)"

This reverts commit 81ff6145516540d7ca2266e4f488754b6c7909ea.

At Alpine we build everything with build type "None" to make sure our
own compiler options are used rather than upstream's. Make sure CMake
allows this.
https://github.com/valhalla/valhalla/pull/2965#issuecomment-821013029

---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 992b7a770..1545304f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,16 +71,16 @@ if(NOT MSVC_IDE) # TODO: May need to be extended for Xcode, CLion, etc.
   endif()
 endif()
 
-if(CMAKE_BUILD_TYPE STREQUAL Debug)
+if(CMAKE_BUILD_TYPE MATCHES Debug)
   message(STATUS "Configuring in debug mode")
-elseif(CMAKE_BUILD_TYPE STREQUAL Release)
+elseif(CMAKE_BUILD_TYPE MATCHES Release)
   message(STATUS "Configuring in release mode")
-elseif(CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo)
+elseif(CMAKE_BUILD_TYPE MATCHES RelWithDebInfo)
   message(STATUS "Configuring in release mode with debug flags")
-elseif(CMAKE_BUILD_TYPE STREQUAL MinRelSize)
+elseif(CMAKE_BUILD_TYPE MATCHES MinRelSize)
   message(STATUS "Configuring in release mode with minimized size")
 else()
-  message(FATAL_ERROR "Unrecognized build type. Use one of Debug, Release, RelWithDebInfo, MinRelSize")
+  message(STATUS "Unrecognized build type - will use cmake defaults")
 endif()
 
 function(create_source_groups prefix)
-- 
2.31.1

